tests/test_workspace.py

```diff
@@ -170,11 +170,16 @@ def test_get_workspace_data(workspace_factory, include_auxdata):
     assert w.data(m, include_auxdata=include_auxdata)
 
 
-def test_get_workspace_data_bad_model(workspace_factory, caplog):
+def test_get_workspace_data_bad_model(workspace_factory, caplog, mocker):
     w = workspace_factory()
     m = w.model()
     # the iconic fragrance of an expected failure
-    m.config.channels = [c.replace('channel', 'chanel') for c in m.config.channels]
+
+    mocker.patch(
+        "pyhf.mixins._ChannelSummaryMixin.channels",
+        new_callable=mocker.PropertyMock,
+        return_value=["channel"],
```

Comment:
```suggestion
        return_value=["chanel"],
```

Hm. pre-commit is pushing back a "fix" here for what it thinks is a `codespell` error (`chanel` -> `channel`), but this is intentional to cause a `KeyError`. So this will need to get ignored somehow, or a different erroneous `return_value` will need to be used.

Reply:
Also the

```python
    with caplog.at_level(logging.INFO, 'pyhf.pdf'):
        with pytest.raises(KeyError):
            assert w.data(m)
            assert 'Invalid channel' in caplog.text
```

test block isn't executing any code after

```python
            assert w.data(m)
```

so the

```python
            assert 'Invalid channel' in caplog.text
```

never gets run. I checked this by putting a `breakpoint` in after

```python
        with pytest.raises(KeyError):
            assert w.data(m)
            breakpoint()
            assert 'Invalid channel' in caplog.text
```

which is never reached.